src/psyclone/tests/test_files/dynamo0p3/testkern_dofs_mod.f90

```diff
@@ -42,26 +42,30 @@ module testkern_dofs_mod
   implicit none
 
   type, extends(kernel_type) :: testkern_dofs_type
-     type(arg_type), dimension(5) :: meta_args =          &
+     type(arg_type), dimension(6) :: meta_args =          &
           (/ arg_type(gh_field, gh_real, gh_write, w1),   &
              arg_type(gh_field, gh_real, gh_read,  w1),   &
              arg_type(gh_field, gh_real, gh_read,  w1),   &
              arg_type(gh_field, gh_real, gh_read,  w1),   &
-             arg_type(gh_scalar, gh_real, gh_read)        &
+             arg_type(gh_scalar, gh_real, gh_read),       &
+             arg_type(gh_field*3, gh_real, gh_read,  &
+              ANY_SPACE_1)                                &
            /)
+
      integer :: operates_on = DOF
    contains
      procedure, nopass :: code => testkern_dofs_code
   end type testkern_dofs_type
 
 contains
 
-  subroutine testkern_dofs_code(a, b, c, d, scalar_arg)
+  subroutine testkern_dofs_code(a, b, c, d, scalar_arg, field_vec)
     implicit none
 
     real(kind=r_def),    intent(inout) :: a
     real(kind=r_def),    intent(in)    :: b, c, d
     real(kind=r_def),    intent(in)    :: scalar_arg
+    type(field_type),    intent(in)    :: field_vec(3)
```

Comment:
A kernel must contain essentially Fortran77 code - i.e. plain arrays rather than objects. Just as `a`, `b`, `c` and `d` are real scalars, this field vector argument should also be three scalars.

Reply:
@mo-alistairp, please have a look at this science kernels that has an example of field vector for coordinate fields (`chi`): https://code.metoffice.gov.uk/trac/lfric/browser/LFRic/trunk/components/science/source/kernel/geometry/sci_calc_detj_at_w2_kernel_mod.F90